src/execution/utils.ts

```diff
@@ -556,6 +570,16 @@ async function signIntent(
   intentOp: IntentOp,
   signers?: SignerSet,
 ) {
+  if (config.account?.type === 'eoa') {
+    if (!config.eoa?.sign) {
+      throw new Error('EOA account must have an EOA configured')
+    }
```

Comment:
I believe this wouldn't work with external wallets like MetaMask, b/c they ban raw signing.

Probably best to use `signTypedData` here.